packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -265,6 +266,22 @@ export function isTaggedTelemetryPropertyValue(x: any): x is ITaggedTelemetryPro
     return (typeof (x?.value) !== "object" && typeof (x?.tag) === "string");
 }
 
+export function parseJSONObject(x: any): TelemetryEventPropertyType | null {
```

Comment:
Can you add a doc comment here?  The type itself explains what it does (coerces the input to `TelemetryEventPropertyType`) but a comment saying the same would be nice.

Also can I suggest a different name? As-is it's pretty generic (and I wouldn't call this "parsing").  Maybe something like `filterOutUnloggableValues`? Or `ensureOnlyValidTelemetryProperties` I might like better.

On the other hand, if you stop exporting it it won't really matter what the name is - I would definitely recommend that!